lib/huginn_acumen_order_agent/concerns/invoice_query_concern.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+# This module is responsible for reading/processing data recrods from the Invoice
+# table in Acumen. This table contains the baseline order/shipment information
+# Shipping method, tracking numbers, etc.
+module InvoiceQueryConcern
+  extend AcumenOrderQueryConcern
+
+  # Fetch/Process the Acumen data,
+  def fetch_invoice_data(acumen_client, order_codes)
+    invoice_data = acumen_client.get_invoices(order_codes)
+
+    return process_invoice_response(invoice_data)
+  end
+
+  # This function returns an array of Acumen invoices.
+  def process_invoice_response(raw_data)
+    raw_data.map do |i|
+
+      log('------------------------------------------------------------')
+          log(":: raw invoice #{i}")
+      log('------------------------------------------------------------')
+
+      invoice = nil
+      begin
+        invoice = response_mapper(i, {
+          'Invoice.Invoice_ID' => 'identifier',
+          'Invoice.Order_Code' => 'orderNumber',
+          'Invoice.Order_Date' => 'orderDate',
+          'Invoice.Status' => 'orderStatus',
+          'Invoice.Customer_Name' => 'customer',
+          'Invoice.Modified_Date' => 'modified_date',
+        })
+
+        invoice['type'] = '@Order'
+
+        #----------  Parse Tracking Numbers  ----------#
+        invoice['orderDelivery'] = []
+        tracking_numbers = get_field_value(i, 'Invoice.Tracking_Num')
+        tracking_numbers = tracking_numbers.blank? ? [] : tracking_numbers.split(/\s|&#13;/)
```

Comment:
What is this `tracking_numbers.split(/\s|&#13;/) doing?

Specifically within the split? Is that encoded characters from acumen?

Reply:
The tracking number field in Acumen is not always a single number. Sometimes it's  a list of tracking numbers.

And Acumen is archaic, so it doesn't store the data as a CSV or an array or any kind of useful format. It stores it with a delimiter of `&#13;` or an HTML encoded `\r` character.

`\s` matches whitespace characters
`|` is basically an `or`
`&#13;` matches that string of characters exactly.

So the the regex is basically saying `look for any white space character of `&#13;`, and the _split_ call it splitting the string on each match it finds. The end result is an array of tracking numbers.